### PR TITLE
[Enh] Convert SASS to PostCSS

### PR DESCRIPTION
**Pull request by ashraj98**
Notes:
- Replace webpack with Vite
- Remove dist folder
    * Dynamically generated when running yarn publish
- Provide workflow to publish to GitHub package registry
  - Include source CSS and postcss.config.js so that clients can use mixins and other postcss features (see `files` in package.json)
  - See example of workflow here: https://github.com/torchdental/batch-data-loader/actions/workflows/publish.yaml
- Replace SASS vars with CSS vars
- Plugins used:
    * `postcss-import` to support `@import` syntax
    * `postcss-mixins` to replace SASS mixins. Not ideal as this is not a planned CSS standard, but difficult to replace
    * `postcss-nested` to handle nested styling. Note that `postcss-preset-env` also includes a nesting parser, but since this comes first in the import list, this plugin will process the nested styles first
    * `postcss-inline-svg` to replace `node-sass-svg`. These "image" icons are compiled from various sources and ideally they should be removed completely along with the fontawesome 4 fonts in favor of FontAwesome 6, which supports a much larger iconset, multiple styles, typesafe imports and tree shaking of unused icons for optimal bundle size. This plugin requires us to hardcode colors for SVGs since you cannot use css vars in a background-image, which is another reason to use FontAwesome, as the react component has much better color support
    * `postcss-url` to inline non-SVG assets. ~~This may not be necessary~~ This is used to inline SVGs not loaded via `postcss-inline-svg`
    * `postcss-preset-env` to enable stage 2 (the default) CSS specs. See https://preset-env.cssdb.org/features/ for list of features
    * `autoprefixer` to prefix styles that have browser specific variants. Note that this produces warnings about using `start` instead of `flex-start` already, probably should be fixed
    * `cssnano` for minifying the distributed css
- Replace `$icon-color` in SVGs with `class="icon-color"`, which can be targeted by `postcss-inline-svg`
- Update workflow to publish to GitHub pages
- We should probably use Google Fonts to get Open Sans rather and serving our own font files

### CODE REVIEW

**Comment by pyen** [comment]
```
$ npm run preview

> @torchdental/torchbrand@2.0.63 preview /home/user/workspace/torch/torchbrand
> vite preview

failed to load config from /home/user/workspace/torch/torchbrand/vite.config.js
error when starting preview server:
Error: Cannot find module 'node:path'
Require stack:
```

Is there a missing config file?

**Comment by ashraj98** [comment]
> ```
> $ npm run preview
> 
> > @torchdental/torchbrand@2.0.63 preview /home/user/workspace/torch/torchbrand
> > vite preview
> 
> failed to load config from /home/user/workspace/torch/torchbrand/vite.config.js
> error when starting preview server:
> Error: Cannot find module 'node:path'
> Require stack:
> ```
> 
> ~Is there a missing config file?~ _investigating locally_
> 
> NodeJS v14 fails. Switching to v17 works-ish:
> 
> ```
> [vite] Internal server error: Failed to resolve import "react/jsx-dev-runtime" from "react/jsx-dev-runtime". Does the file exist?
>   Plugin: vite:import-analysis
>   File: react/jsx-dev-runtime
>   1  |  import * as jsxRuntime from "react/jsx-dev-runtime"
> ```

I was using `yarn dev` but `yarn preview` also works for me on `v14.18.3`:
```
ashwin@Ashwins-Work-MacBook torchbrand % yarn preview
yarn run v1.22.19
$ vite preview
  ➜  Local:   http://127.0.0.1:4173/
  ➜  Network: use --host to expose
^C
ashwin@Ashwins-Work-MacBook torchbrand % node --version
v14.18.3
```

File: src/components/button-group.css [hunk 1]
```diff
@@ -45,27 +45,27 @@
...
     }
     &:first-child {
       border-radius: 4px 0 0 4px; 
     }
     &:last-child {
       border-radius: 0 4px 4px 0; 
     }
-    @include mobile-breakpoint {
+    @mixin mobile-breakpoint {
       align-items: start;
```
Comment (pyen) [inline src/components/button-group.css:69]: May as well patch this to `flex-start`

File: src/components/button-group.css [hunk 2]
```diff
@@ -83,15 +83,15 @@
...
         margin-bottom: 210px;
       }
     }
   }
   
   .buttonGroup .menu--nested {
-    @include tablet-breakpoint {
+    @mixin tablet-breakpoint {
       display: flex;
       align-items: start;
```
Comment (pyen) [inline src/components/button-group.css:96]: May as well patch this to `flex-start`

File: src/components/product.css [hunk 3]
```diff
@@ -257,7 +257,7 @@
     }
   }
 
-  @include mobile-breakpoint {
+  @mixin mobile-breakpoint {
```
Comment (pyen) [inline src/components/product.css:260]: May as patch to `flex-start` for `product-prices` on 265.

File: src/components/nav.css [hunk 4]
```diff
@@ -250,7 +250,7 @@
               top: 0;
             }
             &-item {
-              padding-left: $s32;
+              padding-left: var(--s32);
               display: flex;
               flex-direction: row;
               align-items: start;
```
Comment (pyen) [inline src/components/nav.css:256]: May as well patch this to `flex-start`

**Review by pyen — APPROVED** [review]

**Comment by pyen** [comment]
Only real concern is timing. We would want to have `torchweb/static` updated alongside this PR. Provided we don't wait too long, we can simply hold off on `torchbrand` design changes until both of the prior are complete.

**Pull request merged**